compute/resource_cluster.go

```diff
@@ -321,6 +324,12 @@ func modifyClusterRequest(clusterModel *Cluster) {
 		}
 		clusterModel.AwsAttributes = &awsAttributes
 	}
+	if clusterModel.AzureAttributes != nil {
+		clusterModel.AzureAttributes = nil
```

Comment:
nit: can we add a unit test here

Reply:
But does it make sense to nil the field if it's not nil?..

Reply:
all Azure attributes now are related to the Spot instances only, and if we're using instance pools, then we shouldn't override instance types?